### ISSUE TITLE
tags in samconfig.toml

### ISSUE THREAD

**Issue by superwese** [aws/aws-sam-cli#2559]
This is related to https://github.com/aws/aws-sam-cli/issues/2089 
### Describe your idea/feature/enhancement

It would be nice if tags in samconfig.toml could be passed as a list like parameter_overrides, instead of a long space separated string.
It would make the config file easier to read and to maintain.

### Proposal

add the ability to provide tags as list in samconfig.toml:

```toml
tags = [
    "stage=int",
    "company:application=awesome-service",
    "company:department=sales"
    ]
```

Things to consider:
1. Will this require any updates to the [SAM Spec](https://github.com/awslabs/serverless-application-model)

probably yes. At least a type check in the parser is required

**Comment by mgrandis** [comment]
Thanks @superwese for the suggestion.
As we already have this feature working using the string format, I tagged this as nice to have.
This could be a good first issue for a contributor though.
If anyone wants to implement this, keep in mind that we will have to support both formats.

**Comment by wchengru** [comment]
This is released in SAM CLI 1.25.0
